### ISSUE TITLE
Need longer timeout and retries when probing instance for mount support

### ISSUE THREAD

**Issue by townsend2010** [canonical/multipass#3232]
When an instance is finished booting, it's possible it's very busy starting and running services.  As such, when `multipassd` probes the instance to determine if mount support is enabled, etc., the instance may be so busy that the probing may time out.  We should increase the timeout and/or add some retries.

---
_Original message:_
I am still having this issue with
```
multipass   1.12.2
multipassd  1.12.2
```
[mp.log](https://github.com/canonical/multipass/files/12654425/mp.log)

_Originally posted by @sed-i in https://github.com/canonical/multipass/issues/2986#issuecomment-1724664169_

**Comment by townsend2010** [comment]
Hi @sed-i!

After looking through the log, I see that there was an ssh timeout when the mount handling code was trying to probe the instance to make sure related software was installed.  Since it got a failure, it removed the mount.  Does trying to add the mount again work?

**Comment by sed-i** [comment]
> Does trying to add the mount again work?

Absolutely. More often than not, after the host reboots I need to manually mount. And it works right away.

**Comment by townsend2010** [comment]
Ok.  I only see one time in the log of where it was removed.  Would it possible to get more cases of this so I can be sure of the exact scenario when this occurs?  Also, do you happen to have many services running inside the instances that bog it down and cause it to not respond in a timely manner?  I'm just wondering if these instances are busy on start.

**Comment by townsend2010** [comment]
Ok, so we should probably add a longer timeout and/or be a bit more persistent in trying again a few times.

**Comment by gl-yziquel** [comment]
Can't the timeout be configured somewhere ?
